tests/pipeline/utils.py

```diff
@@ -3,6 +3,7 @@
 import pytest
 import random
 from os import environ
+import io
```

Comment:
i pulled this over from my filesystem state branch because i needed those changes here.